sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -697,4 +705,179 @@ func (s *NetworkFlowManagerTestSuite) TestExpireMessage() {
 	m.Stop(nil)
 }
 
+func TestSendNetworkFlows(t *testing.T) {
+	t.Setenv(features.SensorCapturesIntermediateEvents.EnvVar(), "true")
+	suite.Run(t, new(sendNetflowsSuite))
+}
+
+type sendNetflowsSuite struct {
+	suite.Suite
+	mockCtrl     *gomock.Controller
+	mockEntity   *mocksManager.MockEntityStore
+	m            *networkFlowManager
+	mockDetector *mocksDetector.MockDetector
+	fakeTicker   chan time.Time
+}
+
+const (
+	srcID = "src-id"
+	dstID = "dst-id"
+)
+
+func (b *sendNetflowsSuite) SetupTest() {
+	b.mockCtrl = gomock.NewController(b.T())
+	b.m, b.mockEntity, _, b.mockDetector = createManager(b.mockCtrl)
+
+	b.fakeTicker = make(chan time.Time)
+	go b.m.enrichConnections(b.fakeTicker)
+}
+
+func (b *sendNetflowsSuite) TeardownTest() {
+	b.m.done.Signal()
+}
+
+func (b *sendNetflowsSuite) updateConn(pair *connectionPair) {
+	addHostConnection(b.m, createHostnameConnections("hostname").withConnectionPair(pair))
+}
+
+func (b *sendNetflowsSuite) expectLookups(n int) {
+	expectEntityLookupContainerHelper(b.mockEntity, n, clusterentities.ContainerMetadata{
```

Comment:
To fix the data race after the rebase.
```suggestion
func (b *sendNetflowsSuite) expectLookups(n int) {
    b.mockEntity.EXPECT().RecordTick().AnyTimes()
	expectEntityLookupContainerHelper(b.mockEntity, n, clusterentities.ContainerMetadata{
```